src/lock_free/mwcas_descriptor.cpp

```diff
@@ -69,29 +70,43 @@ constexpr uint64_t kCntMask = (kMwCASFlag - 1UL) ^ (kPosMask | kAddrMask);
 /// @brief A bitmask with only the "MwCAS FLAG" and "descriptor address" portions set to 1.
 constexpr uint64_t kDescMask = kMwCASFlag | kAddrMask;
 
+}  // namespace
+
 /*##############################################################################
- * Local global variable
+ * Static utilities
  *############################################################################*/
 
-/// @brief A thread local descriptor for reuse.
-thread_local std::unique_ptr<MwCASDescriptor> _tls = nullptr;  // NOLINT
+void
+MwCASDescriptor::StartGC(  //
+    const size_t gc_interval,
+    const size_t gc_thread_num)
+{
+  _gc = std::make_unique<EpochBasedGC>(gc_interval, gc_thread_num, kMaxReusableDescriptors);
+}
 
-}  // namespace
+void
+MwCASDescriptor::StopGC()
+{
+  _gc.reset();
+}
+
+auto
+MwCASDescriptor::CreateEpochGuard()  //
+    -> ::dbgroup::thread::EpochGuard
+{
+  return _gc->CreateEpochGuard();
+}
 
 /*##############################################################################
  * Public APIs
  *############################################################################*/
 
 auto
-MwCASDescriptor::GetDescriptor()  //
-    -> MwCASDescriptor *
+MwCASDescriptor::GetDescriptor() -> MwCASDescriptor *
 {
-  auto *desc = _tls.release();
-  if (!desc) {
-    desc = ::dbgroup::memory::Allocate<MwCASDescriptor>();
-  }
+  auto *page = _gc->GetPageIfPossible<MwCASDescriptor>();
+  auto *desc = (page == nullptr) ? new MwCASDescriptor{} : static_cast<MwCASDescriptor *>(page);
```

Comment:
最近はポインタは直にbool値として判定でいいかと思ってます。

```suggestion
  auto *desc = (page) ? static_cast<MwCASDescriptor *>(page) : new MwCASDescriptor{};
```